### PR TITLE
TRUNK-3787: Refactored ConceptDAO

### PR DESCRIPTION
**Pull request by Pilipino**
## Description of what I changed

Hi! This is a new pull request, such that the branch is now named after the ticket id.

I have updated all functions with respect to the Java code conventions, such as including the Javadoc comments on top of public functions. However, getConceptUuids and getConceptReferenceTermsBySource is not included in ConceptService. I was not able to properly document these functions, given that these are not documented in ConceptService. Please advise. Thanks :)

## Issue I worked on

See https://issues.openmrs.org/browse/TRUNK-3787
See https://codein.withgoogle.com/dashboard/task-instances/4852850187304960/

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [ ] I have **added tests** to cover my changes. 
(If you refactored existing code that was well tested you do not have to add tests)

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [ ] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/ConceptDAO.java [hunk 1]
```diff
@@ -120,6 +120,22 @@
...
+	 * @param excludeClasses List&lt;ConceptClass&gt; to leave out of results
+	 * @param requireDatatypes List&lt;ConceptDatatype&gt; to restrict to
+	 * @param excludeDatatypes List&lt;ConceptDatatype&gt; to leave out of results
+	 * @param answersToConcept all results will be a possible answer to this concept
+	 * @return the number of concepts matching the given search phrase
+	 * @throws APIException
+	 * @since 1.8
+	 * @should return a count of unique concepts
+	 */
 	public Integer getCountOfConcepts(String phrase, List<Locale> locales, boolean includeRetired,
```
Comment (lluismf) [inline api/src/main/java/org/openmrs/api/db/ConceptDAO.java]: Too many parameters. Consider using a Search DTO instead. There are some services/DAOs using this pattern.

Comment (Pilipino) [reply api/src/main/java/org/openmrs/api/db/ConceptDAO.java]: @lluismf Would you mean the `@see` javadocs instead? :)

Comment (lluismf) [reply api/src/main/java/org/openmrs/api/db/ConceptDAO.java]: No, I mean replacing the N parameters with a single one.

Comment (Pilipino) [reply api/src/main/java/org/openmrs/api/db/ConceptDAO.java]: Sorry, could you please elaborate? :D

**Comment by Pilipino** [comment]
Sorry, I think there are too many commits on this one. I'll close it.

**Pull request closed without merging**